packages/nodes-base/nodes/SplitInBatches/v4/SplitInBatchesV4.node.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable n8n-nodes-base/node-filename-against-convention */
+import type {
+	IExecuteFunctions,
+	INodeExecutionData,
+	INodeType,
+	INodeTypeDescription,
+} from 'n8n-workflow';
+import { NodeConnectionType, NodeOperationError, deepCopy } from 'n8n-workflow';
+import { getPairedItemInformation } from './utils';
+import type { SplitInBatchesContext } from './types';
+
+export class SplitInBatchesV4 implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'Loop Over Items (Split in Batches)',
+		name: 'splitInBatches',
+		icon: 'fa:sync',
+		group: ['organization'],
+		version: 4,
+		description: 'Split data into batches and iterate over each batch',
+		defaults: {
+			name: 'Loop Over Items',
+			color: '#007755',
+		},
+		// eslint-disable-next-line n8n-nodes-base/node-class-description-inputs-wrong-regular-node
+		inputs: [
+			{ type: NodeConnectionType.Main },
+			{ type: NodeConnectionType.Main, maxConnections: 1 },
+		],
+		inputNames: ['input', 'loop'],
```

Comment:
Probably best we define the names already in "inputs" instead of having "inputs" and "inputNames" defined especially if "inputs" is already using the new object format.